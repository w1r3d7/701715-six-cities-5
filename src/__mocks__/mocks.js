export const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

export const FilterType = {
  POPULAR: `Popular`,
  PRICE_LOW_TO_HIGH: `Price: low to high`,
  PRICE_HIGH_TO_LOW: `Price: high to low`,
  TOP_RATED_FIRST: `Top rated first`,
};

export const offers = [{
  city: {
    name: `Cologne`,
    location: {
      latitude: 50.938361,
      longitude: 6.959974,
      zoom: 13
    }
  },
  title: `The house among olive `,
  rating: 3.8,
  type: `house`,
  bedrooms: 3,
  price: 493,
  description: `The house among olive `,
  location: {
    latitude: 50.957361,
    longitude: 6.9509739999999995,
    zoom: 16
  },
  id: 1,
  bedroomsCount: 3,
  facilities: [
    `Laptop friendly workspace`,
    `Washer`,
    `Towels`,
    `Fridge`,
    `Air conditioning`,
    `Baby seat`,
    `Breakfast`
  ],
  photosUrl: [
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/9.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/10.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/4.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/2.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/11.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/1.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/17.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/5.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`
  ],
  isPremium: false,
  isInBookmark: false,
  hostName: `Angelina`,
  hostDescription: `Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.`,
  hostAvatar: `img/avatar-angelina.jpg`,
  isHostPremium: true,
  maxCapacity: 4,
  previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg`
},
{
  city: {
    name: `Brussels`,
    location: {
      latitude: 50.846557,
      longitude: 4.351697,
      zoom: 13
    }
  },
  title: `Amazing and Extremely Central Flat`,
  rating: 4.3,
  type: `apartment`,
  bedrooms: 3,
  price: 323,
  description: `Amazing and Extremely Central Flat`,
  location: {
    latitude: 50.839557,
    longitude: 4.346697,
    zoom: 16
  },
  id: 2,
  bedroomsCount: 3,
  facilities: [
    `Towels`,
    `Baby seat`,
    `Laptop friendly workspace`,
    `Dishwasher`,
    `Washing machine`,
    `Air conditioning`,
    `Washer`,
    `Coffee machine`,
    `Breakfast`,
    `Fridge`
  ],
  photosUrl: [
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/10.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/11.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/20.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/7.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/18.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/8.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/9.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/4.jpg`
  ],
  isPremium: false,
  isInBookmark: false,
  hostName: `Angelina`,
  hostDescription: `This is a place for dreamers to reset, reflect, and create. Designed with a 'slow' pace in mind, our hope is that you enjoy every part of your stay; from making local coffee by drip in the morning, choosing the perfect record to put on as the sun sets.`,
  hostAvatar: `img/avatar-angelina.jpg`,
  isHostPremium: true,
  maxCapacity: 8,
  previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/8.jpg`
},
{
  city: {
    name: `Brussels`,
    location: {
      latitude: 50.846557,
      longitude: 4.351697,
      zoom: 13
    }
  },
  title: `The house among olive `,
  rating: 4.6,
  type: `apartment`,
  bedrooms: 4,
  price: 218,
  description: `The house among olive `,
  location: {
    latitude: 50.865556999999995,
    longitude: 4.371696999999999,
    zoom: 16
  },
  id: 3,
  bedroomsCount: 4,
  facilities: [
    `Washer`,
    `Laptop friendly workspace`,
    `Breakfast`,
    `Air conditioning`,
    `Baby seat`
  ],
  photosUrl: [
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/2.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/1.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/12.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/17.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/5.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/11.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/6.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/4.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/8.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/7.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/10.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg`
  ],
  isPremium: false,
  isInBookmark: false,
  hostName: `Angelina`,
  hostDescription: `Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country. In our beautiful screened Pondhouse, you can gaze at the stars and listen to the sounds of nature from your cozy warm bed.`,
  hostAvatar: `img/avatar-angelina.jpg`,
  isHostPremium: true,
  maxCapacity: 7,
  previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg`
}
];


export const reviews = [{
  "id": 1,
  "rating": 4,
  "date": `2020-11-15T13:58:19.494Z`,
  "reviewText": `421421`,
  "photoUrl": `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/3.jpg`,
  "name": `xyz`,
  "isUserPremium": false,
  "userId": 1
}, {
  "id": 2,
  "rating": 3,
  "date": `2020-10-15T13:58:19.494Z`,
  "reviewText": `1111`,
  "photoUrl": `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/4.jpg`,
  "name": `xyza`,
  "isUserPremium": true,
  "userId": 2
}
];
