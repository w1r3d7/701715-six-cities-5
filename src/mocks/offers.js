const getRating = () => {
  return String((Math.random() * 5).toFixed(2));
};

const getRandomItemsCount = (array) => {
  const shuffled = array.sort(() => 0.5 - Math.random());
  return shuffled.slice(0, Math.floor(Math.random() * array.length));
};

const reviews = [
  {
    name: `Oleg`,
    id: 11,
    photoUrl: `/img/avatar-max.jpg`,
    reviewText: `Lorem ipsum dolor sit amet, consectetur adipisicing elit. Incidunt laudantium tempore veritatis? A
      asperiores debitis deserunt maxime molestiae nihil obcaecati, odio officiis quos repellendus sed, sit ut veniam?
      Eligendi, illo. `,
    rating: getRating(),
    date: new Date().toDateString(),
  },
  {
    name: `Petr`,
    id: 21,
    photoUrl: `/img/avatar-max.jpg`,
    reviewText: `Lorem ipsum dolor sit amet, consectetur adipisicing elit. Incidunt laudantium tempore veritatis? A
      asperiores debitis deserunt maxime molestiae nihil obcaecati, odio officiis quos repellendus sed, sit ut veniam?
      Eligendi, illo. `,
    rating: getRating(),
    date: new Date().toDateString(),
  },
  {
    name: `Max`,
    id: 31,
    photoUrl: `/img/avatar-max.jpg`,
    reviewText: `Lorem ipsum dolor sit amet, consectetur adipisicing elit. Incidunt laudantium tempore veritatis? A
      asperiores debitis deserunt maxime molestiae nihil obcaecati, odio officiis quos repellendus sed, sit ut veniam?
      Eligendi, illo. `,
    rating: getRating(),
    date: new Date().toDateString(),
  },
  {
    name: `Angelina`,
    id: 41,
    photoUrl: `/img/avatar-angelina.jpg`,
    reviewText: `Lorem ipsum dolor sit amet, consectetur adipisicing elit. Incidunt laudantium tempore veritatis? A
      asperiores debitis deserunt maxime molestiae nihil obcaecati, odio officiis quos repellendus sed, sit ut veniam?
      Eligendi, illo. `,
    rating: getRating(),
    date: new Date().toDateString(),
  },
];

export default [
  {
    id: 1,
    city: `Amsterdam`,
    photosUrl: [`/img/apartment-01.jpg`, `/img/apartment-small-03.jpg`, `/img/apartment-small-04.jpg`],
    type: `Apartment`,
    price: 120,
    description: `Beautiful & luxurious studio at great location`,
    isPremium: true,
    isInBookmark: false,
    facilities: [
      `Wi-Fi`,
      `Towels`,
      `Washing machine`,
      `Heating`,
    ],
    rating: getRating(),
    bedroomsCount: 3,
    maxCapacity: 4,
    hostName: `Angelina`,
    hostAvatar: `/img/avatar-angelina.jpg`,
    isHostPremium: true,
    hostDescription: `Lorem ipsum dolor sit amet, consectetur adipisicing elit. Dolor eius et facilis fuga mollitia
      praesentium quasi qui quibusdam quo recusandae repellat, repellendus rerum saepe sapiente sequi unde velit vero
      voluptatem.`,
    reviews: getRandomItemsCount(reviews),
  },
  {
    id: 2,
    city: `Amsterdam`,
    photosUrl: [`/img/apartment-02.jpg`, `/img/apartment-small-03.jpg`, `/img/apartment-small-04.jpg`],
    type: `House`,
    price: 40,
    description: `Great location`,
    isPremium: true,
    isInBookmark: true,
    facilities: [
      `Wi-Fi`,
      `Towels`,
      `Washing machine`,
      `Heating`,
    ],
    rating: getRating(),
    bedroomsCount: 1,
    maxCapacity: 2,
    hostName: `Igor`,
    hostAvatar: `/img/avatar-max.jpg`,
    isHostPremium: false,
    hostDescription: `Lorem ipsum dolor sit amet, consectetur adipisicing elit. Blanditiis doloremque in iure
      laborum magnam placeat quasi quidem quo! Assumenda error fuga iste laborum molestiae nam odit quia repellendus sed
      soluta.`,
    reviews: getRandomItemsCount(reviews),
  },
  {
    id: 3,
    city: `Amsterdam`,
    photosUrl: [`/img/apartment-03.jpg`, `/img/apartment-small-03.jpg`, `/img/apartment-small-04.jpg`],
    type: `Villa`,
    price: 4000,
    description: `Nice, cozy, warm big bed apartment`,
    isPremium: false,
    isInBookmark: false,
    facilities: [
      `Washing machine`,
      `Towels`,
      `Heating`,
      `Coffee machine`,
      `Baby seat`,
    ],
    rating: getRating(),
    bedroomsCount: 4,
    maxCapacity: 5,
    hostName: `Oleg`,
    hostAvatar: `/img/avatar-max.jpg`,
    isHostPremium: true,
    hostDescription: `Lorem ipsum dolor sit amet, consectetur adipisicing elit. Blanditiis doloremque in iure
      laborum magnam placeat quasi quidem quo! Assumenda error fuga iste laborum molestiae nam odit quia repellendus sed
      soluta.`,
    reviews: getRandomItemsCount(reviews),
  },
  {
    id: 4,
    city: `Amsterdam`,
    photosUrl: [`/img/room.jpg`, `/img/apartment-small-03.jpg`, `/img/apartment-small-04.jpg`],
    type: `Private Room`,
    price: 4,
    description: `Wood and stone place`,
    isPremium: false,
    isInBookmark: true,
    facilities: [
      `Kitchen`,
      `Dishwasher`,
      `Cabel TV`,
      `Fridge`,
    ],
    rating: getRating(),
    bedroomsCount: 1,
    maxCapacity: 1,
    hostName: `Oleg`,
    hostAvatar: `/img/avatar-max.jpg`,
    isHostPremium: false,
    hostDescription: `Lorem ipsum dolor sit amet, consectetur adipisicing elit. Blanditiis doloremque in iure
      laborum magnam placeat quasi quidem quo! Assumenda error fuga iste laborum molestiae nam odit quia repellendus sed
      soluta.`,
    reviews: getRandomItemsCount(reviews),
  },
];
