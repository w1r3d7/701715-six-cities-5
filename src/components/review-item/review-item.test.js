import React from 'react';
import renderer from 'react-test-renderer';

import ReviewItem from './review-item';

import {reviews} from '../../__mocks__/mocks';

const [review] = reviews;

test(`ReviewItem render correctly`, () => {
  const tree = renderer
    .create(
        <ReviewItem review={review} />
    ).toJSON();

  expect(tree).toMatchSnapshot();
});


