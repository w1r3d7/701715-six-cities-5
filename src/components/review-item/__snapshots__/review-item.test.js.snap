// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewItem render correctly 1`] = `
<li
  className="reviews__item"
>
  <div
    className="reviews__user user"
  >
    <div
      className="reviews__avatar-wrapper user__avatar-wrapper"
    >
      <img
        alt="Avatar xyz"
        className="reviews__avatar user__avatar"
        height="54"
        src="https://assets.htmlacademy.ru/intensives/javascript-3/avatar/3.jpg"
        width="54"
      />
    </div>
    <span
      className="reviews__user-name"
    >
      xyz
    </span>
  </div>
  <div
    className="reviews__info"
  >
    <div
      className="reviews__rating rating"
    >
      <div
        className="reviews__stars rating__stars"
      >
        <span
          style={
            Object {
              "width": "80%",
            }
          }
        />
        <span
          className="visually-hidden"
        >
          Rating
        </span>
      </div>
    </div>
    <p
      className="reviews__text"
    >
      421421
    </p>
    <time
      className="reviews__time"
      dateTime="2020-11-15T13:58:19.494Z"
    >
      November 2020
    </time>
  </div>
</li>
`;
